components/member_list_channel/member_list_channel.jsx

```diff
@@ -144,14 +145,20 @@ export default class MemberListChannel extends React.Component {
                 const user = users[i];
 
                 if (teamMembers[user.id] && channelMembers[user.id] && user.delete_at === 0) {
-                    usersToDisplay.push(user);
+                    const status = UserStore.getStatus(user.id);
+                    usersToDisplay.push({...user, status});
+
                     actionUserProps[user.id] = {
                         channel: this.props.channel,
                         teamMember: teamMembers[user.id],
                         channelMember: channelMembers[user.id]
                     };
                 }
             }
+
+            usersToDisplay.sort((a, b) => {
+                return UserStatusesWeight[a.status] - UserStatusesWeight[b.status] || Utils.sortUsersByDisplayName(a, b);
```

Comment:
I don't quite understand this - is it some kind of type coercion trick? It looks like you're subtracting the weighted values of the two users' statuses, and if that results in zero, we know that zero is falsy, so evaluate the OR condition, which sorts by name?

Reply:
I guess it's short-circuit evaluation. Your subsequent description is what I intend to do which I  think works fine as I expected.